neurodamus/report.py

```diff
@@ -0,0 +1,259 @@
+import logging
+from .core import NeurodamusCore as Nd
+
+
+def get_section_index(cell, section):
+    """
+    Calculate the global index of a given section within its cell.
+    :param cell: The cell instance containing the section of interest
+    :param section: The specific section for which the index is required
+    :return: The global index of the section, applicable for neuron mapping
+    """
+    section_name = str(section)
+    base_offset = 0
+    section_index = 0
+    if "soma" in section_name:
+        pass  # base_offset is 0
+    elif "axon" in section_name:
+        base_offset = cell.nSecSoma
+    elif "dend" in section_name:
+        base_offset = cell.nSecSoma + cell.nSecAxonalOrig
+    elif "apic" in section_name:
+        base_offset = cell.nSecSoma + cell.nSecAxonalOrig + cell.nSecBasal
+    elif "ais" in section_name:
+        base_offset = cell.nSecSoma + cell.nSecAxonalOrig + cell.nSecBasal + cell.nSecApical
+    elif "node" in section_name:
+        base_offset = cell.nSecSoma + cell.nSecAxonalOrig + cell.nSecBasal + cell.nSecApical \
+                    + getattr(cell, 'nSecLastAIS', 0)
+    elif "myelin" in section_name:
+        base_offset = cell.nSecSoma + cell.nSecAxonalOrig + cell.nSecBasal + cell.nSecApical \
+                    + getattr(cell, 'nSecLastAIS', 0) + getattr(cell, 'nSecNodal', 0)
+
+    # Extract the index from the section name
+    try:
+        index_str = section_name.split('[')[-1].rstrip(']')
+        section_index = int(index_str)
+    except ValueError:
+        logging.warning(f"Error while getting section index {index_str}")
+
+    return int(base_offset + section_index)
+
+
+class Report:
+    INTRINSIC_CURRENTS = {"i_membrane", "i_membrane_", "ina", "ica", "ik", "i_pas", "i_cap",
+                          "ihcn_Ih", "ihcn_hcn3", "icsa_csa", "reportcai_mcd"}
+
+    def __init__(self, report_name, report_type, variable_name, unit, format, dt, start_time,
+                 end_time, output_dir, electrode=None, scaling_option=None, isc_param=None):
+        self.report_name = report_name
+        self.variable_name = variable_name
+        self.report_dt = dt
+        self.scaling_mode = self.determine_scaling_mode(scaling_option)
+
+        self.alu_list = []
+        self.report = Nd.SonataReport(0.5, report_name, output_dir, start_time,
+                                      end_time, dt, unit, "compartment")
+        Nd.BBSaveState().ignore(self.report)
+
+    def determine_scaling_mode(self, scaling_option):
+        if scaling_option is None or scaling_option == 'Area':
+            return 1  # SCALING_AREA
+        elif scaling_option == 'None':
+            return 0  # SCALING_NONE
+        else:
+            return 2  # SCALING_ELECTRODE
+
+    def add_compartment_report(self, cell_obj, point, vgid, use_coreneuron=False,
+                                pop_name="default", pop_offset=0):
+        if use_coreneuron:
```

Comment:
Can this be an object setting, rather than being queried / passed through for every cell?